code/modules/mob/living/carbon/human/ego_gifts.dm

```diff
@@ -72,11 +72,7 @@
 			locked = locked ? FALSE : TRUE
 			owner.ShowGifts()
 		if("hide")
-			if(visible)
-				owner.cut_overlay(mutable_appearance(src.icon, src.icon_state, src.layer))
-			else
-				owner.add_overlay(mutable_appearance(src.icon, src.icon_state, src.layer))
-			visible = !visible
+			Refresh_Gift_Sprite(visible) //for uniquely colored gifts
```

Comment:
Probably want to mention this in the PR

Reply:
Oh shit thats true. I cant really atomize it from parasite tree since parasite tree is the first iteration of its usage.